dvc/remote/base.py

```diff
@@ -680,20 +680,33 @@ def safe_remove(self, path_info, force=False):
         self.remove(path_info)
 
     def _checkout_file(
-        self, path_info, checksum, force, progress_callback=None
+        self,
+        path_info,
+        checksum,
+        force,
+        progress_callback=None,
+        save_link=True,
     ):
-        cache_info = self.checksum_to_path_info(checksum)
-        if self.exists(path_info):
-            msg = "data '{}' exists. Removing before checkout."
-            logger.warning(msg.format(str(path_info)))
+        if self._needs_checkout(path_info, checksum):
             self.safe_remove(path_info, force=force)
 
-        self.link(cache_info, path_info)
-        self.state.save_link(path_info)
-        self.state.save(path_info, checksum)
+            cache_info = self.checksum_to_path_info(checksum)
+            self.link(cache_info, path_info)
+            if save_link:
+                self.state.save_link(path_info)
+            self.state.save(path_info, checksum)
+        else:
+            if self.protected:
```

Comment:
Let's leave a comment telling why we are just protecting/unprotecting and not checking the current state, it is not that obvious.

Reply:
done